react/components/EXPERIMENTAL_Table/hooks/useTableVisibility.ts

```diff
@@ -32,13 +29,19 @@ const useHiddenColumns = (columns: Array<Column>) => {
   }
 
   return {
-    hiddenColumns,
-    setHiddenColumns,
+    columns,
     visibleColumns,
+    hiddenColumns,
     toggleColumn,
     showAllColumns,
     hideAllColumns,
   }
 }
 
-export default useHiddenColumns
+export type VisibilityData = {
+  columns: Array<Column>
+}
+
+function getHiddenColumns(columns: Array<Column>): Array<string> {
+  return columns.filter(col => col.hidden).map(col => col.id)
```

Comment:
I'm curious... Can we use [flatMap](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap)? It would be awesome to use it here!

Reply:
Done